PD-PiotrKurzydym/praca-domowaJS-nr1-Piotr-Kurzydym.js

```diff
@@ -0,0 +1,605 @@
+          getFavouriteColor: '' // funkcja zamiast pustego stringa
+      },
+      OUTPUT
+      {
+          Bartolomeo: "firstName",
+          Lozano: "lastName",
+          Rabona: "nickname",
+      },
+      e) zredukuj tablicę obiektów do pojedynczego obiektu, który będzie zawierał wszystkie klucze i wartości
+      wszystkich obiektów z tablicy, dzięki temu, że w punkcie d) odwrócilismy klucze z wartościami, nie będzie 
+      z tym problemu :)
+      *f) odtwórz z obiektu tablicę, która będzie zawierała same nicknames i 
+      nazwiska, ktdjest < s i
+      imię, którego chociaż jedna litera jest większa >= s
+      g) posortuj tablicę alfabetycznie
+  */
+console.log(`Zadanie 6.a)`);
+const multi = (a) => {
+  return (b) => {
+    return a * b;
+  };
+};
+
+console.log(multi(5)(6));
+const multiplyBySix = multi(6);
+console.log(multiplyBySix(10));
+console.log(`---`);
+
+console.log(`Zadanie 6.b)`);
+const multi2 = (a) => {
+  return (b) => {
+    return (c) => {
+      return (d) => {
+        return a * b * c * d;
+      };
+    };
+  };
+};
+console.log(multi2(4)(5)(6)(10));
+
+console.log(`---`);
+/*
+      *6. Currying function
+      a) Napisz taką funkcję mnożącą 2 liczby, aby możliwe były następujące wywołania:
+      - multi(5)(6)
+      - const multiplyBySix = sum(6)
+        multiplyBySix(10)
+      b) Analogicznie napisz funkcję, która mnoży 4 liczby i możliwe jest wywołanie
+      - multi(4)(5)(6)(10)
+      Hints:
+      - funkcja może zwracać inne funkcje
+      - funkcja może korzystać ze zmiennych i parametrów funkcji zewnętrznych (czyli tych w których się znajduje)
+  */
+
+/*
+      **7. Rekurencja
+       a) Mając zagnieżdżony obiekt, wyciągnij z niego wszystkie imiona i dodaj do tablicy
+       ***b) Jeżeli osoba ma więcej niż jedno imię, te imiona powinny znajdować się w jednym stringu w tablicy
+       Na przykład 'Kamil Bartek'
+      INPUT:
+  */
+console.log(`Zadanie 7.`);
+
+const nestedObject = {
+  name: "Kamil",
+  children: [
+    {
+      name: "Zosia",
+    },
+    {
+      name: "Krysia",
+      name2: "Barbara",
+      children: [
+        {
+          name: "Basia",
+          children: [
+            {
+              name: "Monika",
+              name2: "Viola",
+              children: [
+                {
+                  name: "Mateusz",
+                },
+                {
+                  name: "Sebastian",
+                  name2: "August",
+                  name3: "Franciszek",
+                  children: [
+                    { name: "Alex" },
+                    { name: "Stasio" },
+                    {
+                      name: "Paulina",
+                      children: [{ name: "Kuba" }, { name: "Kacper" }],
+                    },
+                  ],
+                },
+              ],
+            },
+          ],
+        },
+      ],
+    },
+  ],
+};
+let names = [];
+
+function getNames(items) {
```

Comment:
To nie jest funkcja napisana w sposób rekurencyjny. Ideą rekurencji jet to że funkcja sama siebie wywołuje. A tu wewnątrz ciała funkcji nigdzie nie ma jej wywołania. Nie mniej szacun za próbę i kawał kodu :).